clients/js/src/auth.ts

```diff
@@ -0,0 +1,229 @@
+        }
+        this.credentialsProvider = options.credentialsProvider || new BasicAuthCredentialsProvider(options.textCredentials);//new BasicAuthCredentialsProvider(options.textCredentials || options.credentialsProvider);
+    }
+
+    authenticate(): ClientAuthResponse {
+        return new BasicAuthClientAuthResponse(this.credentialsProvider.getCredentials());
+    }
+}
+
+export class IsomorphicFetchClientAuthProtocolAdapter implements ClientAuthProtocolAdapter<RequestInit> {
+    authProvider: ClientAuthProvider | undefined;
+    wrapperApi: DefaultApi | undefined;
+
+    /**
+     * Creates a new adapter of IsomorphicFetchClientAuthProtocolAdapter.
+     * @param api - The API to wrap.
+     * @param authConfiguration - The configuration for the authentication provider.
+     */
+
+    constructor(private api: DefaultApi, authConfiguration: AuthOptions) {
+
+        switch (authConfiguration.provider) {
+            case "basic":
+                this.authProvider = new BasicAuthClientAuthProvider({textCredentials: authConfiguration.credentials, credentialsProvider: authConfiguration.credentialsProvider});
+                break;
+            default:
+                this.authProvider = undefined;
+                break;
+        }
+        if (this.authProvider !== undefined) {
+            this.wrapperApi = this.wrapMethods(this.api);
+        }
+    }
+
+    getApi(): DefaultApi {
+        return this.wrapperApi ?? this.api;
+    }
+
+    getAllMethods(obj: any): string[] {
+        let methods: string[] = [];
+        let currentObj = obj;
+
+        do {
+            const objMethods = Object.getOwnPropertyNames(currentObj)
+                .filter(name => typeof currentObj[name] === 'function' && name !== 'constructor');
+
+            methods = methods.concat(objMethods);
+            currentObj = Object.getPrototypeOf(currentObj);
+        } while (currentObj);
+
+        return methods;
+    }
+
+    wrapMethods(obj: any): any {
+        let self = this;
+        const methodNames = Object.getOwnPropertyNames(Object.getPrototypeOf(obj))
+            .filter(name => typeof obj[name] === 'function' && name !== 'constructor');
+
+        return new Proxy(obj, {
+            get(target, prop: string) {
+                if (methodNames.includes(prop)) {
+                    return new Proxy(target[prop], {
+                        apply(fn, thisArg, args) {
+                            const modifiedArgs = args.map(arg => {
+                                if (arg && typeof arg === 'object' && 'method' in arg) {
+                                    return self.injectCredentials(arg as RequestInit);
+                                }
+                                return arg;
+                            });
+                            if (Object.keys(modifiedArgs[modifiedArgs.length - 1]).length === 0) {
+                                // modifiedArgs.push({} as RequestInit);
```

Comment:
not needed?

Reply:
Absolutely not :). Gonna remove it. As you see I have a tendency to ignore things I have commented out (just like interpreted/complier would). I need jslint or SonarQube to hint me about this.